labs/sysdev-u-boot-bbb/sysdev-u-boot-bbb.tex

```diff
@@ -377,20 +379,42 @@ \subsection{Network configuration on the PC host}
 
 \section{Setting up the TFTP server}
 
-Let's install a TFTP server on your development workstation:
+Let's install a TFTP (Trivial File Transfer Protocol) server on your development workstation:
 
 \begin{verbatim}
 sudo apt install tftpd-hpa
 \end{verbatim}
 
 You can then test the TFTP connection. First, put a small text file in
-the directory exported through TFTP on your development
+the directory exported through TFTP (\code{/srv/tftp} by default) on your development
 workstation. Then, from U-Boot, do:
 
 \begin{ubootinput}
 => tftp %\zimageboardaddr% textfile.txt
 \end{ubootinput}
 
+If the loading was not succesfull, e.g.:
+\begin{verbatim}
+=> tftp 0x81000000 textfile.txt
+high speed config #2: 2 mA, Ethernet Gadget, using RNDIS
+USB RNDIS network up!
+Using usb_ether device
+TFTP from server 192.168.0.1; our IP address is 192.168.0.100
+Filename 'textfile.txt'.
+Load address: 0x81000000
+Loading: T T T T T T T T T T T T T T T T T T T T
+Retry count exceeded; starting again
+\end{verbatim}
+make sure that the TFTP port(\code{69} by default) is open on the firewall:
+\begin{verbatim}
+sudo ufw allow 69/udp
+\end{verbatim}
+
+It is advised to open all ports for communication, since there will be other servers involved:
+\begin{verbatim}
+sudo ufw allow from 192.168.0.100
+\end{verbatim}
```

Comment:
Ah, here is the TFTP access. Just like for NFS, I don't like the "It is advised to open all ports for communication, since there will be other servers involved". Either we open ports, or we allow all traffice from 192.168.0.100, but doing both doesn't make sense.